### DISCUSSION TITLE
Survey: New Algorithms for OptunaHub

### DISCUSSION THREAD

**Discussion by toshihikoyanase** [optuna/optuna#5606] [Ideas]
We’re excited to announce an open call for algorithms to be included in [OptunaHub](https://hub.optuna.org/)! This discussion thread aims to expand our library and ensure we effectively meet the community's needs. Here’s how you can contribute:

- Please comment below with the algorithm(s) you want to OptunaHub. 
- If someone has already suggested the same algorithm you were thinking of, please upvote their comment instead of posting a duplicate suggestion.

This will help us prioritize the most popular algorithms and streamline decision-making. Your participation is crucial to OptunaHub's continual growth and improvement, and we deeply appreciate your contributions.

Thank you!

**Comment by toshihikoyanase** [comment]
Differential Evolution
Reference: Storn, R., Price, K. Differential Evolution – A Simple and Efficient Heuristic for Global Optimization over Continuous Spaces. Journal of Global Optimization 11, 341–359 (1997). https://doi.org/10.1023/A:1008202821328

**Comment by nzw0301** [comment]
Optimizing Millions of Hyperparameters by Implicit Differentiation, Jonathan Lorraine, Paul Vicol, David Duvenaud. In _AISTATS_, 108:1540-1552, 2020. https://proceedings.mlr.press/v108/lorraine20a.html.

**Comment by mfeurer** [comment]
I would love to see SMAC3's HPO facade and Google Vizier.

**Reply by y0z** [reply]
Thank you for your feedback.
SMAC3 is available now on OptunaHub!
- https://hub.optuna.org/samplers/smac_sampler/
- example: https://github.com/optuna/optunahub-registry/blob/main/package/samplers/smac_sampler/example.py

<a href="https://hub.optuna.org/samplers/smac_sampler/"><img width="1350" alt="image" src="https://github.com/user-attachments/assets/132cbbdb-e633-4613-ab69-79b9758b91f9"></a>

**Comment by okaikov** [comment]
Motivation
implement a new BO algorithm to leverage user expert knowledge to reduce sample size needed for convergence

Description
Leveraging Domain Information for the Efficient Automated Design of Deep Learning Accelerators - https://www.cs.utexas.edu/~lin/papers/hpca23.pdf

in the above paper is mentioned a new algorithm for leveraging user expert knowledge to define new features to help the BO model to learn more quickly.

**Comment by jt269** [comment]
Do samplers hosted on Optuna Hub support distributed optimisation?

**Reply by nzw0301** [reply]
I suppose it depends on the implementation.

**Comment by rodrigoscoelho** [comment]
TuRBO

**Reply by y0z** [reply]
I created an issue that requests a TuRBO package.
https://github.com/optuna/optunahub-registry/issues/186

**Reply by y0z** [reply]
TuRBO is now available on OptunaHub.
https://hub.optuna.org/samplers/turbo/

**Reply by rodrigoscoelho** [reply]
great. thanks
